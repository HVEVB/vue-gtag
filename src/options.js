import { reactive, toRefs } from "vue";

export const options = reactive({
  id: null,
  params: {},
  isGtagEnabled: false,
  globalObjectName: "gtag",
  globalDataLayerName: "dataLayer",
  domain: "https://www.googletagmanager.com",
  customResource: null,
});

export const useOptions = () => {
  return toRefs(options);
};

export const mergeOptions = (newOptions = {}) => {
  Object.keys(newOptions).forEach((key) => {
    options[key] = newOptions[key];
  });
};
