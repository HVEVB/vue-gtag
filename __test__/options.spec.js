import { options } from "@/options";

describe("options", () => {
  it("should render default plugin options", () => {
    expect(options).toMatchSnapshot();
  });
});
