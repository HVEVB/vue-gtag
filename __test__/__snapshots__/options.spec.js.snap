// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`options should render default plugin options 1`] = `
Object {
  "customResource": null,
  "domain": "https://www.googletagmanager.com",
  "globalDataLayerName": "dataLayer",
  "globalObjectName": "gtag",
  "id": null,
  "isGtagEnabled": false,
  "params": Object {},
}
`;
